ska_trend/wrong_box_anom/wrong_box_anom.py

```diff
@@ -0,0 +1,367 @@
+        np.abs(other_slots["zang"][closest_slot] - anom_row["zang_obs"])
+        < (other_slots["halfw"][closest_slot] + BOX_PAD)
+    ):
+        dat.update(
+            {
+                "classic": True,
+                "actual_slot": other_slots["slot"][closest_slot],
+                "actual_slot_idd": other_slots["acqid"][closest_slot],
+                "actual_slot_mag_obs": other_slots["mag_obs"][closest_slot],
+            }
+        )
+
+    # Get star catalog for the obsid
+    sc = get_starcheck_catalog(anom_row["obsid"])
+
+    # And get the RA and Dec of what was tracked (cdy cdz puts things approx at the
+    # target location)
+    y = anom_row["yang"] + anom_row["cdy"]
+    z = anom_row["zang"] + anom_row["cdz"]
+
+    # Check if the acquired star was actually an acquisition star
+    intended = acqs[
+        (acqs["obsid"] == anom_row["obsid"]) & (acqs["slot"] != anom_row["slot"])
+    ]
+    is_another_acq_star = np.any(
+        (np.abs(intended["yang"] - y) < 7) & (np.abs(intended["zang"] - z) < 7)
+    )
+    dat.update({"another_acq_star": is_another_acq_star})
+
+    # Check if an agasc star was tracked
+    ra, dec = yagzag2radec(
+        y / 3600.0,
+        z / 3600.0,
+        Quat((sc["obs"]["point_ra"], sc["obs"]["point_dec"], sc["obs"]["point_roll"])),
+    )
+    stars = agasc.get_agasc_cone(
+        ra, dec, radius=7.0 / 3600.0, date=anom_row["guide_tstart"]
+    )
+    dat.update(
+        {
+            "agasc_star": (len(stars) > 0)
+            & (np.any(stars["MAG_ACA"] <= (anom_row["mag_obs"] + 1.5)))
+        }
+    )
+    return dat
+
+
+def acq_anom_checks(acqs):
+    """
+    Perform anomaly checks on acquisition observations.
+
+    Parameters
+    ----------
+    acqs : astropy.table.Table
+        The table of acquisition observations.
+
+    Returns
+    -------
+    astropy.table.Table
+        The table of acquisition observations with additional anomaly information.
+    """
+    anom = wrong_box(acqs)
+    anoms = acqs[anom]
+    new_info = [get_anom_info(anom_row, acqs) for anom_row in anoms]
+    new_info = Table(new_info)
+    anoms = hstack([anoms, new_info])
+    return anoms
+
+
+def add_manual_entries(anom_table):
+    """
+    Add manual entries to the anomaly table.
+
+    The data is just hardcoded within this function as for this trending application one can edit
+    the file and update quickly via non-fsds process.
+
+    Parameters
+    ----------
+    anom_table : astropy.table.Table
+        The table of anomaly data.
+
+    Returns
+    -------
+    astropy.table.Table
+        The table of anomaly data with manual entries added.
+    """
+
+    table = anom_table.copy()
+    table = table[
+        "obsid",
+        "slot",
+        "classic",
+        "acq_start",
+        "guide_tstart",
+        "mag_obs",
+        "actual_slot",
+        "actual_slot_idd",
+        "actual_slot_mag_obs",
+        "another_acq_star",
+        "agasc_star",
+    ]
+
+    manual_rows = [
```

Comment:
Let's move away from putting stuff like this in Python code in the package. For this particular case we could make a supplementary manual data file that lives in `$SKA/www/ASPECT/wrong_box_anom` which we edit as needed. Given how these data are used, that level of process control is adequate.

The action here is capture this thought as a code comment here. The NEXT time you come to the code to add a new manual event, instead change the code to store manual events in an ECSV file. You can easily create it using this code.

Reply:
Right, that's why I brought this up.  Though I note that I took more than a few minutes to find the manual data for kalman_watch3 even with the README file that somebody left around as breadcrumbs -- so I'm not sure if package data (if not Python code) for non-fsds packages might end up being easier to manage.